File: src/main/java/com/mexchange/api/entity/Transaction.java
package com.mexchange.api.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.util.Date;

@NoArgsConstructor
@Getter
@Setter
@Document(collection = "transactions")
public class Transaction {
    private User user;
    private User.Recipient recipient;
    Date date;
    String originCurrency;
    String targetCurrency;
    BigDecimal amount;
    BigDecimal exchangeRate;
    String status;
}

File: src/main/kotlin/com/mxchange/midrate/api/entity/Transaction.kt
package com.mxchange.midrate.api.entity

import org.springframework.data.mongodb.core.mapping.Document
import java.math.BigDecimal
import java.util.*

@Document(collection = "transactions")
data class Transaction (
    var user: User? = null,
    var recipient: User.Recipient? = null,
    var date: Date? = null,
    var originCurrency: String? = null,
    var targetCurrency: String? = null,
    var amount: BigDecimal? = null,
    var exchangeRate: BigDecimal? = null,
    var status: String? = null
)
